consensus/istanbul/core/core.go

```diff
@@ -769,14 +772,18 @@ func (c *core) verifyProposal(proposal istanbul.Proposal) (time.Duration, error)
 	} else {
 		logger.Trace("verification status cache miss")
 
-		duration, err := c.backend.Verify(proposal)
+		result, duration, err := c.backend.Verify(proposal)
 		logger.Trace("proposal verify return values", "duration", duration, "err", err)
 
 		// Don't cache the verification status if it's a future block
 		if err != consensus.ErrFutureBlock {
 			c.current.SetProposalVerificationStatus(proposal.Hash(), err)
 		}
 
+		if result != nil {
```

Comment:
it seems to be that if `err == nil` then `result != nil`. If that doesn't hold, it's a bug. 

So my suggestion is to either: 
 a) remove the if here. 
 b) replace it by checking if `result == nil` and returning an error there, clearly stating it's a bug.

Reply:
`err` could be an Error other than ErrFutureBlock, in which case `if result != nil` is necessary